### ISSUE TITLE
Parse error on JSON file with strings that contain semicolons

### ISSUE THREAD

**Issue by iwehrman** [jshint/jshint#2488]
```
$ cat test.json
{
    "a": ";"
}

$ jshint test.json 
test.json: line 2, col 5, Expected an assignment or function call and instead saw an expression.
test.json: line 2, col 8, Missing semicolon.
test.json: line 2, col 8, Expected '}' to match '{' from line 1 and instead saw ':'.
test.json: line 2, col 10, Expected an assignment or function call and instead saw an expression.
test.json: line 2, col 13, Missing semicolon.
test.json: line 2, col 1, Unrecoverable syntax error. (50% scanned).

6 errors

$ jshint --version
jshint v2.8.0
```

**Comment by caitp** [comment]
This will be a big where we need to use checkPunctuators rather than just testing the value of the token,i believe. Should be an easy first big if anyone's interested

**Comment by jugglinmike** [comment]
Yup, I'm halfway there :)

**Comment by caitp** [comment]
I said *first* bug :p

**Comment by nicolo-ribaudo** [comment]
Now this can be closed

**Comment by lukeapage** [comment]
This will be released in the next version of jshint, after 2.8.0
